### DISCUSSION TITLE
Add possibility to extend the Grafana menus from plugins

### DISCUSSION THREAD

**Discussion by mckn** [grafana/grafana#44004] [Plugins]
This is a break out feature request from the following issue: https://github.com/grafana/grafana/issues/31381

It would be nice to be able to extend the following menus with new options from plugins:

- Add possibility to extend the dashboard panel menu with links into the app plugin.
- Add possibility to extend the plugin menu with external links.

**Comment by torkelo** [comment]
Would need a lot of detail on what types of panels should have this added? (panel plugin type, maybe setting condition, data condition? )

**Comment by matryer** [comment]
Grafana Incident has a use case for this.

We'd like to offer a "Declare Incident" option (and even "Add to Incident"), which would allow users to attach the panel as context to an incident.

* We would highlight the panel in the activity timeline
* We'd use data from the panel (the queries etc.) to influence any automated checks we might perform
* So we'd need a way to learn as much as we can about a panel, including which dashboard it was from, in which folder, etc. which I suppose is possible via the API?

Additionally, it would also be nice to indicate that a panel is a part of an active incident, with a glowing fire outline, or by showing the incident icon on the panel, but that might be a step too far for a first version of this.

**Comment by mckn** [comment]
@grafana/dashboards-squad is this something you would like to have input on?

**Comment by mckn** [comment]
This will be solved by the following epic issue: https://github.com/grafana/grafana/issues/61658

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
